homeassistant/components/yamaha/media_player.py

```diff
@@ -147,7 +151,35 @@ def service_handler(service):
             device.schedule_update_ha_state(True)
 
     hass.services.register(
-        DOMAIN, SERVICE_ENABLE_OUTPUT, service_handler, schema=ENABLE_OUTPUT_SCHEMA
+        DOMAIN,
+        SERVICE_ENABLE_OUTPUT,
+        service_handler_enable_output,
+        schema=ENABLE_OUTPUT_SCHEMA,
+    )
+
+    def service_handler_select_service(service):
+        """Handle for services."""
+        entity_ids = service.data.get(ATTR_ENTITY_ID)
+
+        _LOGGER.info("Select Scene: %s %s", entity_ids, service.data[ATTR_SCENE])
+
+        devices = [
+            device
+            for device in hass.data[DATA_YAMAHA].values()
+            if not entity_ids or device.entity_id in entity_ids
+        ]
+
+        for device in devices:
+            scene = service.data[ATTR_SCENE]
+
+            device.select_scene(scene)
+            device.schedule_update_ha_state(True)
+
+    hass.services.register(
```

Comment:
Please refactor and use our entity service helper instead.

https://developers.home-assistant.io/docs/dev_101_services#entity-services

Reply:
Hey, did that. But for that the whole discovery had to be migrated. Learned a lot :)